### PR TITLE
[Docs] Python API finer documentation

### PR DESCRIPTION
**Pull request by Kathryn-cat**
See progress in 
- https://github.com/mlc-ai/mlc-llm/issues/682

Co-author: @CharlieFRuan

### CODE REVIEW

File: python/mlc_chat/chat_module.py [hunk 1]
```diff
@@ -519,6 +519,8 @@ def __init__(
             ``mlc-chat-config.json``.
         lib_path : Optional[str]
             The full path to the model library file to use (e.g. a ``.so`` file).
+            If unspecified, we will use the provided ``model`` to searhc over
```
Comment (MasterJH5574) [inline python/mlc_chat/chat_module.py:522]:
```suggestion
            If unspecified, we will use the provided ``model`` to search over
```

File: docs/deploy/python.rst [hunk 2]
```diff
@@ -21,16 +21,187 @@ Verify Installation
...
+      Hello! I'm here to help you with your question. However, I must point out
+      ...
+
+      runtime_stats='prefill: 336.0 tok/s, decode: 118.5 tok/s'
+
+|
+
+.. note:: 
+   You could also specify the address of ``model`` and ``lib_path`` explicitly. If
+   you only specify ``model`` as ``model_name`` and ``quantize_moed``, we will
```
Comment (MasterJH5574) [inline docs/deploy/python.rst:123]: `mod` I guess?
```suggestion
   you only specify ``model`` as ``model_name`` and ``quantize_mod``, we will
```

Comment (CharlieFRuan) [reply docs/deploy/python.rst:123]: Ohh `mode`!

File: docs/deploy/python.rst [hunk 3]
```diff
@@ -21,16 +21,187 @@ Verify Installation
...
+
+After making sure that the files exist, from the ``mlc-llm`` directory, we can
+run the following code with the conda environment you used to install ``mlc_chat``:
+
+.. code:: python
+
+   from mlc_chat.chat_module  import ChatModule
+   cm = ChatModule(model='Llama-2-7b-chat-hf-q4f16_1')  # Create a ChatModule instance
+   cm.generate(prompt="What is the meaning of life?") # Generate a response for a given prompt
+   runtime_stats = cm.runtime_stats_text()  # Print some runtime statistics for the generation
```
Comment (MasterJH5574) [inline docs/deploy/python.rst:101]: `cm.runtime_stats_text()` was renamed to `cm.stats()` in #688. The wheel build are in progress and will be ready shortly.

File: docs/deploy/python.rst (hunk 2 above)
Comment (MasterJH5574) [reply docs/deploy/python.rst:123]: Ah you’re right

**Review by tqchen — APPROVED** [review]

**Comment by Kathryn-cat** [comment]
Finishing Try Out MLC Chat section now. Please do not merge.

**Review by MasterJH5574 — APPROVED** [review]

**Comment by MasterJH5574** [comment]
Merged the doc page for Python API. The try-out part will be in a separate PR.

**Pull request merged**